### ISSUE TITLE
How to plan for a sequence of target poses using motion_planning_api

### ISSUE THREAD

**Issue by kanster** [ros-planning/moveit#1005]
### Description

Using move_group, computeCartesianPath allows the users to plan a trajectory for a sequence of targeting poses. However, what if the target poses do not follow this pattern, how to plan a trajectory which visits all the targeting poses?

Similar questions have been raised in previous google groups and ROS discussion board:
https://groups.google.com/forum/#!topic/moveit-users/lZL2HTjLu-k
https://answers.ros.org/question/296994/how-to-set-a-sequence-of-goals-in-moveit/
However, there is no clear answer to this.

This is what I have tried with motion planning request and response api:
```
    std::vector<double> tolerance_pose(3, 0.01);
    std::vector<double> tolerance_angle(3, 0.01);

    for ( int i = 0; i <= nposes; ++ i ) {
        geometry_msgs::PoseStamped pose;
        pose.header.frame_id = "base_link";
        pose.pose.position.x = i*0.1+line_minx;
        pose.pose.position.y = line_y;
        pose.pose.position.z = line_z;
        pose.pose.orientation.w = 1.0;
        req.group_name = "manipulator";
        moveit_msgs::Constraints pose_goal =
                kinematic_constraints::constructGoalConstraints("wrist_3_link", pose, tolerance_pose, tolerance_angle);
        req.goal_constraints.push_back(pose_goal);
    }
```
However, the multiple targets included in the request is not processed by context solve function.
By investigating the moveit and ompl source code, we are looking at `ompl::base::PlannerInputStates::nextGoal(const PlannerTerminationCondition &ptc)` in _planner.cpp_ and `ompl_interface::GoalSampleableRegionMux::maxSampleCount()` from _goal_union.cpp_. 

In short, only one _State_ is returned to the `ompl::geometric::RRTConnect::solve(const base::PlannerTerminationCondition &ptc)` function and it __seems__ that other goals are not processed.

Is there any function in moveit which allows such operation?

Br
K

### Your environment
* ROS Distro: Kinetic
* OS Version: Ubuntu 16.04
* Source: kinetic-devel
